src/Sylius/Bundle/ApiBundle/Doctrine/ORM/QueryExtension/Shop/Order/ShopUserBasedExtension.php

```diff
@@ -11,26 +11,82 @@
 
 declare(strict_types=1);
 
-namespace Sylius\Bundle\ApiBundle\Doctrine\QueryItemExtension;
+namespace Sylius\Bundle\ApiBundle\Doctrine\ORM\QueryExtension\Shop\Order;
 
+use ApiPlatform\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
 use ApiPlatform\Doctrine\Orm\Extension\QueryItemExtensionInterface;
 use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
 use ApiPlatform\Metadata\Operation;
 use Doctrine\ORM\QueryBuilder;
 use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
 use Sylius\Bundle\ApiBundle\Serializer\ContextKeys;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
 use Sylius\Component\Core\Model\OrderInterface;
 use Sylius\Component\Core\Model\ShopUserInterface;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
 
-final readonly class OrderShopUserItemExtension implements QueryItemExtensionInterface
+final readonly class ShopUserBasedExtension implements QueryCollectionExtensionInterface, QueryItemExtensionInterface
 {
+    /**
+     * @param array<array-key, mixed> $nonFilteredCartAllowedOperations
+     */
     public function __construct(
         private UserContextInterface $userContext,
         private array $nonFilteredCartAllowedOperations = [],
     ) {
     }
 
+    /**
+     * @param array<array-key, mixed> $context
+     */
+    public function applyToCollection(
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        ?Operation $operation = null,
+        array $context = [],
+    ): void {
+        if (!is_a($resourceClass, OrderInterface::class, true)) {
+            return;
+        }
```

Comment:
Why don't you use the same private method for both collection and item methods?

Reply:
Since it wasn't the focus of this PR, I've tried to minimize the amount of logic changes unless the content of item/collection methods were 1:1; these two differ a bit so I didn't touch them.